packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -82,12 +82,26 @@ export interface EngineReportingOptions<TContext> {
    */
   reportErrorFunction?: (err: Error) => void;
   /**
-   * A case-sensitive list of names of variables whose values should not be sent
-   * to Apollo servers, or 'true' to leave out all variables. In the former
+   * [TO-BE-DEPRECATED] A case-sensitive list of names of variables whose values should
+   * not be sent to Apollo servers, or 'true' to leave out all variables. In the former
    * case, the report will indicate that each private variable was redacted; in
    * the latter case, no variables are sent at all.
    */
   privateVariables?: Array<String> | boolean;
+  /**
+   * A case-sensitive list of names of variables whose values should
+   * not be sent to Apollo servers, a custom function for modifying variable values,
+   * or 'true' v. 'false' to blacklist v. whitelist all variables, respectively. In the first
+   * case, the report will indicate each private variable redacted; in the second,
+   * the report will indicate that variable values were modified by a custom function;
+   * in the final case, the report will indicate whether the variables were all
+   * blacklisted or whitelisted. Will default to 'true' if not specified, and
+   * the report will indicate that no privacy enforcer was provided.
+   */
+  enforcePrivateVariables?:
```

Comment:
Might this be more maintainable as an object? For instance, something like

```ts
enforcePrivateVariables: {
  always: boolean;
  privateVariableNames: Array<String>;
  variableModifier: ((v: Record<string, any>) => Record<string, any>)
}
```

Reply:
w.r.t types, I'm still curious about your thoughts from earlier on whether we should only allow users the ability to **filter** variables (or selectively redact) or if we should let them modify in whatever way they want.

I'm tempted toward the approach you chose here, since it means that users can now obfuscate their personal information / variables, which seems like a huge win. Even so, maybe there is a middle ground, where the key values need to remain the same, even though the values can be modified. Just wanted to bring it up for discussion in case you had more thoughts.

Reply:
@zionts Keeping the key values the same makes sense to me, unless some people need to include a different or larger set of variables in the trace?

In this case I think I'd want to have 'variableModifier' as some function that operates on individual keys (e.g. (v: string) => string or (v: string) => any) to simplify any cleanup/error handling that we would have to do if we just accepted a Record<string, any>, and have the user take care of handling cases.. what do you think?

Reply:
@zionts RE your first suggestion though: I still need to read up on TypeScript, but in this case would it be possible to specify all three fields at once? In this case, would we also need to think about hierarchy between the three fields (e.g. boolean > custom function > array)?

Reply:
Yep, that's right. Or we could just specify that if more than one is set, we'll throw an error? Or we can make the type 
```
{ always: boolean } | { privateVariableNames: Array<String> } | { variableModifier: ... }
```
🤷‍♂ idk

Reply:
I like this ^ idea!